### PR TITLE
fix: All property control renders when JS toggle button is clicked

### LINKED ISSUE #11040: [Type]-[18352]:All the Property pane controls re-renders when we press js toggle
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

_No response_

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

TL;DR: Fix the performance issue that happens when we click the JS toggle button in property pane.

Each property controls were getting rendered whenever we click the JS toggle button. This is since one of the selector used in PropertyControl was returning `dynamicPropertyPathList` which will always change when we click JS toggle button anywhere. Now, instead of returning `dynamicPropertyPathList` from that selector, we are returning a derived boolean value that is specific for the PropertyControl which calls this selector.
Similar behaviour was noticed with `dynamicTriggerPathList` as well. That is also resolved in a similar manner.

While working on this, I noticed the ActionCreator was also rendering since we aren't using proper selectors. That is also fixed.

Introduced [proxy-momoize](https://github.com/dai-shi/proxy-memoize) library which is an alternative to reselect's createSelector. This makes use of ES6 proxies to do an efficient memoization.

Fixes #11040

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by SatishGandham — DISMISSED** [review]
LGTM except that one minor change requested.

Good job with the PR description :)

**Comment by SatishGandham** [comment]
@aswathkk , any performance numbers?

**Comment by aswathkk** [comment]
@SatishGandham I found the scripting time reduced from ~50ms to ~25ms after clicking on the JS toggle.
